shark_turbine/kernel/wave/constraints.py

```diff
@@ -79,6 +79,14 @@ def linearized_thread_id(self) -> IndexExpr:
         ]
         return sum([x * y for x, y in zip(thread_ids, threads_per_block)])
 
+    # Inline substitution for vector_size given index map. In the future we can add support for other members.
+    def _subs(self, index_map: dict[IndexSymbol, int]):
+        if not isinstance(self.vector_shapes, dict):
```

Comment:
Isnt the type of vector shapes always dict? So when would we hit this case?

Reply:
Some tests hit this case for some reason, that's why I added it there

Reply:
btw @harsh-nod, this is because if we specify MMA_type i.e (`mma_type=tkw.MMAType.F32_16x16x16_F16`), in HW constraints, we won't really be using `vector_shapes`. as also shown in the implementation here:
https://github.com/iree-org/iree-turbine/blob/1870973138768bbae4d278ade46782bdde63bcd4/shark_turbine/kernel/wave/expansion.py#L410-L415

Reply:
Makes sense. So if vector_shapes are not defined, are they set to None and can we check for that? 
```
if self.vector_shapes is None:
  return
```

The other thing is that this function is meant to convert the vector_shapes from their symbolic types to static shapes. Could we rename to either subs_vector_shapes (if we want to restrict the scope for now) or finalize (which is what IndexingContext uses, this implies its more general and we don't have to specify what is being substituted)

Reply:
Yeah that makes sense, in the future we'd probably also let `HardwareConstraint::subs` to call multiple of these `subs_member_variable`. :)